pkg/kubelet/server/server.go

```diff
@@ -878,6 +891,63 @@ func (s *Server) getPortForward(request *restful.Request, response *restful.Resp
 	proxyStream(response.ResponseWriter, request.Request, url)
 }
 
+// checkpoint handles the checkpoint API request. It checks if the requested
+// podNamespace, pod and container actually exist and only then calls out
+// to the runtime to actually checkpoint the container.
+func (s *Server) checkpoint(request *restful.Request, response *restful.Response) {
+	if request == nil || response == nil {
+		return
+	}
+	options := &runtimeapi.CheckpointContainerRequest{}
+	if request.Request.ContentLength > 0 {
```

Comment:
Not sure if this is needed either. Can we make the parameter extraction similar to the other handlers for consistency?

Reply:
I switched to query parameters as part of the URL as seen in exec, run, tail.